python/ray/experimental/tqdm_ray.py

```diff
@@ -177,7 +177,6 @@ def __init__(self, state: ProgressBarState, pos_offset: int):
             desc=state["desc"] + " " + str(state["pos"]),
             total=state["total"],
             position=pos_offset + state["pos"],
-            leave=False,
```

Comment:
however, this has the effect of leaving the progress bar in the output after completion:
![Screenshot at Jun 26 19-55-24](https://github.com/ray-project/ray/assets/5122851/74f8edbe-2ced-4b23-a8cf-b4a2b1fe0508)

personally, i don't think it's a bad outcome to leave the progress bar output after completion, as it makes it easy for me to track overall progress. but happy to hear from others, or if there are users who prefer to hide it after completion by default

Reply:
I think it'd be useful if we persisted progress bars, but can we configure this at the Ray Data level? If we change the configuration here, it'd affect other libraries that use `tqdm_ray`.

Btw, is this necessary to fix the issue, or is this an orthogonal change?

Reply:
without this change, the progress bars would disappear after completion, which is the intended behavior when setting `leave=False` in `tqdm`: https://tqdm.github.io/docs/tqdm/#tqdm-objects 

agree that it makes sense to have some configuration which can be set separately. i am thinking i can add a `hide_progress_bars_after_completion` parameter to `tqdm_ray`, which can be set from Ray Data's `ProgressBar` class

Reply:
Sorry, still a bit confused -- if we just made the  `process.update_bar(state)` change and didn't modify `leave`, would we still be able to close the issue?

Reply:
if we did not modify `leave`, there is another issue where all of the progress bars will go away after completion, due to `leave=False` being set (this is the current behavior). The fix in this PR causes the bars to update and disappear after the bar completes:
<img width="1389" alt="Screenshot at Jun 27 11-38-13" src="https://github.com/ray-project/ray/assets/5122851/8eeeab84-19bc-4192-a603-1946f5110094">

so to account for this new issue, we can have `leave=True` to keep the bars around.

Reply:
Ah, gotcha. Yeah, that looks janky

Don't have a strong opinion about whether we `leave=True` for just Jupyter or for all Ray Data progress bars. @raulchen what's your opinion?

Let's definitely add a configuration for `leave` in `tqdm_ray.tqdm` and only configure for Ray Data. Want to avoid changing the behavior for other libraries and users who might use `tqdm_ray`.

Reply:
personally I actually prefer always leaving the progress bars. because it's convenient to inspect some stats. 
One minor concern is that it might be confusing for some users, making them the pipeline still running. 
What about we update the progress bar message in the end to something like `Dataset execution finished in .. seconds`, `Dataset execution failed, see above stack trace for error message`?

Reply:
Got it, thanks for the feedback. Added the functionality to show the result for the overall progress bar as below.

- On success: 
![success](https://github.com/ray-project/ray/assets/5122851/3360e4ca-d8df-4d37-b3a5-efd0d4080af1)

- On failure: 
![failure](https://github.com/ray-project/ray/assets/5122851/bfaf7130-4bd0-4255-94af-ad5301ee936f)

Reply:
Also no longer need to set `leave` in `tqdm_ray` or in Ray Data, since we will always be using `leave=True`, and this is the default value in `tqdm`.